### PR TITLE
Post an idea functionality

### LINKED ISSUE #3: Post idea // 8
In order to get feedback as an idea poster I want to post an idea
- [ ] Idea Model
- [ ] Generate Form
- [ ] Form Handler
- [ ] Validation On Model

### PR DESCRIPTION
**Pull request by ussuritiger**
In this PR I've added idea form, idea form handler and adjusted controller action. It closes #3

### CODE REVIEW

**Comment by piwi91** [comment]
:+1:

File: src/AppBundle/Controller/IdeaController.php [hunk 1]
```diff
@@ -30,9 +34,25 @@ public function viewAction($id)
         ));
     }
 
-    public function createIdeaAction()
+    /**
+     * @param Request $request
+     * @Route("/idea/create", name="idea_create")
```
Comment (dennisdegreef) [inline src/AppBundle/Controller/IdeaController.php:44]: Why is this here? As far as I know, the routing annotations don't work? for reasons i forgot... There is a `routes.yml` in the bundle

File: src/AppBundle/Form/Handler/Idea.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            $idea = new IdeaEntity();
+            $idea
+                ->setTitle($data['title'])
+                ->setDescription($data['description'])
+                ->setUser($this->getUserFromTokenStorage());
+
+            $this->ideaService->saveIdea($idea);
+
+            if ($data['youtube-flag'] === true) {
+                $youtubeLink = new Link();
```
Comment (dennisdegreef) [inline src/AppBundle/Form/Handler/Idea.php:93]: You could create a static factory method for this

File: src/AppBundle/Form/Handler/Idea.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                    ->setType(Link::TYPE_YOUTUBE)
+                    ->setLink($data['youtube-value']);
+                $this->linkService->saveLink($youtubeLink);
+            }
+        }
+    }
+
+    private function getUserFromTokenStorage()
+    {
+        if (($token = $this->tokenStorage->getToken()) !== null) {
```
Comment (dennisdegreef) [inline src/AppBundle/Form/Handler/Idea.php:105]: So this can be a `resource`, `array`, `string` or whatever? :)

I'd say that's an assumption :)
http://rskuipers.com/entry/from-assumptions-to-assertions

File: src/AppBundle/Service/IdeaService.php [hunk 4]
```diff
@@ -35,8 +35,23 @@ public function listIdeas()
         return $this->ideaRepository->findAll();
     }
 
+    /**
+     * @param Idea $idea
+     */
+    public function saveIdea(Idea $idea)
+    {
+        $idea->setCreatedAt(new \DateTime());
```
Comment (dennisdegreef) [inline src/AppBundle/Service/IdeaService.php:54]: If you want to update an idea, it will override the `createdAt` property

File: src/AppBundle/Form/Handler/Idea.php (hunk 3 above)
Comment (piwi91) [inline src/AppBundle/Form/Handler/Idea.php:105]: I already did read that blog post, nice one to remind me to it! I can fix that :)
